src/protocols/tacacs/encode.c

```diff
@@ -23,191 +23,697 @@
  * @copyright 2017 The FreeRADIUS server project
  * @copyright 2017 Network RADIUS SARL (legal@networkradius.com)
  */
-#include <freeradius-devel/util/base.h>
-#include <freeradius-devel/util/net.h>
+
+#include <freeradius-devel/io/test_point.h>
+#include <freeradius-devel/protocol/tacacs/dictionary.h>
 #include <freeradius-devel/server/base.h>
 #include <freeradius-devel/server/log.h>
+#include <freeradius-devel/util/base.h>
 #include <freeradius-devel/util/debug.h>
-#include <freeradius-devel/protocol/tacacs/dictionary.h>
+#include <freeradius-devel/util/net.h>
+#include <freeradius-devel/util/struct.h>
 
 #include "tacacs.h"
 #include "attrs.h"
 
-int fr_tacacs_packet_encode(RADIUS_PACKET * const packet, char const * const secret, UNUSED size_t secret_len)
+/**
+ *	Encode a TACACS+ 'arg_N' fields.
+ */
+static size_t tacacs_encode_body_arg_n_len(VALUE_PAIR *vps, fr_dict_attr_t const *da, uint8_t **body)
```

Comment:
Please convert this and all other encoder functions to use sbuffs or dbuffs depending on which API you think is easier. We won't be accepting new encoder functions that use manual length checks like this.